### ISSUE TITLE
Will this be updated to 5.1?

### ISSUE THREAD

**Issue by jtanta** [doxxx/ffxiv-craft-opt-web#332]
## Required Information

Class: All of them
Level: All 60+
Craftsmanship: n/a 
Control: n/a
CP: n/a
Recipe Name: n/a
Recipe Level: n/a
Solver Seed: (Look at the top of the Execution Log on the Solver page)

 I was just wondering if the Crafting Optimiser will be updated to 5.1. I love this thing!

**Comment by jtanta** [comment]
Thank you so much for the update! Enjoy your vacation!

**Comment by NocturneBlock** [comment]
Vacation is definitely a good reason to wait. Looking forward to the update- this thing is invaluable for preventing crafting implosions. :-)

**Comment by ryan-dyer-sp** [comment]
I have created an initial PR for updating for 5.1.  https://github.com/doxxx/ffxiv-craft-opt-web/pull/333

If anyone plays with it and finds issues, please comment on the PR and I will attempt to address.  Please note that I only have a 55 max crafter, so I was unable to test the higher level changes and relied on asking in my FC for how things worked.

**Comment by jtanta** [comment]
Thank you so much!! I really really appreciate it!!

**Comment by ShammyLevva** [comment]
@ryan-dyer-sp 
My crafter's are 80 in every class so I can perhaps help however have you seen the extent of the changes? This doesn't so much need a PR as a new code branch the changes are that radical.

On logging in I note on my CRP for instance 21 skills no longer available. Replaced by numerous different ones. 

I've created a new issue to list all the changes and suggest the project owner creates a new branch and we mark off the implementation of the changes are processed.

**Comment by ryan-dyer-sp** [comment]
@ShammyLevva Yes I've seen the changes.  Have you seen the PR?  A PR stands for Peer Review and by the way they are implemented requires a new code branch.  

I went through the patch notes 1x1 and made adjustments where I saw appropriate.  Not having a crafter past 50, not all of them were clearly observable to me via login (ie changes around speciailizations).

However, as you have said, there were alot of changes and I'm not perfect, so it is true that I missed a few which people have caught and I have fixed.  And there are some changes which I'm not qualified to implement (final appraisal).

I have not added items to this, as frankly, I dont need them for my personal use of this tool yet.  But if there are some that people who are testing would like me to add, post the stats and I'll get it added to the db.

**Comment by doxxx** [comment]
Thanks for getting this started. We can merge improvements piecemeal as we figure things out.

(PR actually stands for Pull Request, 😉)

**Comment by ShammyLevva** [comment]
@ryan-dyer-sp it's great you've added things the issue is it's difficult to tell from such a large commit which changes you've made and which you've yet to make. Ideally with a massive patch change you make dozens of small commits so each change can be documented as fitting a known requirement of the patch notes. Then as each change is checked it can be applied to the master. 

The bigger the patch you submit the time taken to check it increases exponentially, and the chances of an error/typo slipping through increases. 

It would be useful if you could document which changes from the patch notes you believe you've implemented and which you haven't as that will save hours of work trying to verify your huge patch.

**Comment by jtanta** [comment]
I know this is a dumb question, but what link was the new updates added to? 

Thanks!

**Comment by JaydenWindsong** [comment]
> I have created an initial PR for updating for 5.1. #333
> 
> If anyone plays with it and finds issues, please comment on the PR and I will attempt to address. Please note that I only have a 55 max crafter, so I was unable to test the higher level changes and relied on asking in my FC for how things worked.
> 
> Temp URL: http://ffxivcraftingoptimizer-env.36iwtyrjze.us-east-1.elasticbeanstalk.com:8001/index.html#/solver

Hey, just letting you know that the optimizer at this link doesn't pull the Skybuilder recipes from the new Rebuild Isghard questline.

**Comment by tanpro260196** [comment]
> > I have created an initial PR for updating for 5.1. #333
> > If anyone plays with it and finds issues, please comment on the PR and I will attempt to address. Please note that I only have a 55 max crafter, so I was unable to test the higher level changes and relied on asking in my FC for how things worked.
> > Temp URL: http://ffxivcraftingoptimizer-env.36iwtyrjze.us-east-1.elasticbeanstalk.com:8001/index.html#/solver
> 
> @ryan-dyer-sp, just letting you know that the optimizer at this link doesn't pull the Skybuilder recipes from the new Rebuild Isghard questline.

This PR only adds new actions, not recipes.

**Comment by doxxx** [comment]
I'll be adding new recipes once the 5.1 PR is merged.

**Comment by doxxx** [comment]
The initial set of patch 5.1 changes have been deployed to the beta site. I'll be running the recipe scraper tonight, so there will be updated recipes later today or tomorrow.

**Comment by ShammyLevva** [comment]
Will that scraper pickup the changes to the lower level recipes? In particular weaver recipe changes?

**Comment by doxxx** [comment]
The recipe scraper didn't find anything new, most likely since the recipes were already updated by @kongspark (thanks!). These have already been deployed to the beta site along with the rest of the updates earlier this evening.

Thanks again to @ryan-dyer-sp for doing the lion's share of the work!

**Comment by doxxx** [comment]
@ShammyLevva Could you give an example that I could check?

**Comment by tanpro260196** [comment]
> The initial set of patch 5.1 changes have been deployed to the beta site. I'll be running the recipe scraper tonight, so there will be updated recipes later today or tomorrow.

Ingenuity count seems to be wrong. I got this result for example:

> /ac "Reflect" <wait.3>
/ac "Ingenuity" <wait.2>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Great Strides" <wait.2>
**/ac "Ingenuity" <wait.2>**
/ac "Prudent Touch" <wait.3>
/ac "Great Strides" <wait.2>
/ac "Byregot's Blessing" <wait.3>
/ac "Master's Mend" <wait.3>
**/ac "Ingenuity" <wait.2>**
/echo Macro #1 complete <se.1>

**Comment by doxxx** [comment]
The handling of Ingenuity's countdown hasn't changed. Is there another macro that follows on from that one?

**Comment by tanpro260196** [comment]
> The handling of Ingenuity's countdown hasn't changed. Is there another macro that follows on from that one?

yeah. 

> /ac "Prudent Touch" <wait.3>
/ac "Careful Synthesis" <wait.3>
/ac "Careful Synthesis" <wait.3>
/ac "Careful Synthesis" <wait.3>
/ac "Careful Synthesis" <wait.3>
/echo Macro #2 complete <se.14>

**Comment by ShammyLevva** [comment]
> @ShammyLevva Could you give an example that I could check?

I don’t have a specific example just noticed the patch notes mentioned a reworking of numerous weaver recipes.

**Comment by doxxx** [comment]
@tanpro260196 I think the placement is "random" -- it needed two Ingenuity to cover 9 actions.

Bear in mind that the solver is somewhat random process. You can try running the solver for long to see if it comes up with something better. You can also edit the solver result in the simulator to tweak the ordering/positioning of actions and rerun the solver on that to see if it can improve on it.

**Comment by doxxx** [comment]
@ShammyLevva Let me know if you spot any discrepancies.
